doc/RichTextArea/RichTextArea.md

```diff
@@ -0,0 +1,295 @@
+|PASTE	|Inserts rich text from the clipboard
+|PASTE_PLAIN_TEXT	|Inserts plain text from the clipboard
+|REDO	|Reverts the last undo operation
+|SELECT_ALL	|Selects all text in the document
+|SELECT_DOCUMENT_END	|Selects text (or extends selection) from the current caret position to the end of document
+|SELECT_DOCUMENT_START	|Selects text (or extends selection) from the current caret position to the start of document
+|SELECT_DOWN	|Selects text (or extends selection) from the current caret position one visual text line down
+|SELECT_LEFT	|Selects text (or extends selection) from the current position to one symbol to the left
+|SELECT_PAGE_DOWN	|Selects text (or extends selection) from the current position to one page down
+|SELECT_PAGE_UP	|Selects text (or extends selection) from the current position to one page up
+|SELECT_PARAGRAPH	|Selects text (or extends selection) of the current paragraph
+|SELECT_RIGHT	|Selects text (or extends selection) from the current position to one symbol to the right
+|SELECT_UP	|Selects text (or extends selection) from the current caret position one visual text line up
+|SELECT_WORD	|Selects word at the caret position
+|SELECT_WORD_LEFT	|Extends selection to the previous word (LTR) or next word (RTL)
+|SELECT_WORD_NEXT	|Extends selection to the next word
+|SELECT_WORD_NEXT_END	|Extends selection to the end of next word
+|SELECT_WORD_PREVIOUS	|Extends selection to the previous word
+|SELECT_WORD_RIGHT	|Extends selection to the next word (LTR) or previous word (RTL)
+|UNDO	|Undoes the last edit operation
+
+These functions and the key mappings can be customized using the control's InputMap.
+
+
+
+### Extensibility
+
+RichTextArea is designed with extensibility in mind.  A number of mechanisms are provided for the application developer to customize the control behavior:
+
+- extending the model by adding new attributes
+- adding new functions with new key bindings
+- redefining the existing key bindings
+- by adding left and right side paragraph decorators
+- providing custom scroll bars via **ConfigurationParameters**
+
+The following example illustrates how the basic navigation can be altered to support custom navigation (for example, allowing to jump to the next CamelCase word):
+
+        richTextArea.getInputMap().registerFunction(RichTextArea.MOVE_WORD_NEXT, () -> {
+            // refers to custom logic
+            TextPos p = getCustomNextWordPosition(richTextArea);
+            richTextArea.setCaret(p);
+        });
+
+
+
+## Alternatives
+
+A number of open source projects do exist:
+
+- https://github.com/FXMisc/RichTextFX
+- https://github.com/gluonhq/rich-text-area
+- https://github.com/andy-goryachev/FxEditor
+
+
+## Testing
+
+Two manual test applications are provided - one for RichTextArea (**RichTextAreaDemoApp**)
+and one for the CodeArea (**CodeAreaDemoApp**).
+
+In addition to these two testers, a small example provides a standalone rich text editor, see **RichEditorDemoApp**.
+
```

Comment:
This section should be expanded to include your strategy for how you will test the implementation of the API (both automated and manual testing). I think this can wait, so maybe add a placeholder for now? Something like: "Develop additional automated and manual tests  to validate the implementation."

Reply:
the first sentence describes the strategy, I think.